modules/p2/ding_serendipity/ding_serendipity.module

```diff
@@ -350,6 +352,29 @@ function ding_serendipity_fetch(array $context = NULL) {
 }
 
 /**
+ * Flush serendipity cache of one user.
+ *
+ * @param int $uid
+ *   The user uid.
+ */
+function ding_serendipity_user_cache_clear($uid=NULL) {
+  if ($uid === NULL) {
+    $uid = $GLOBALS['user']->uid;
+  }
```

Comment:
I would much prefer if the parameter to this function was not optional.

1. Making it optional forces us to rely on globals. We not not want to do that. 
2. The only case where the function is used could just as well pass the uid as a parameter.
3. Not making it optional also means we do not have to document behavior if the parameter is omitted in the DocBlock (which is not the case either now)